warehouse/schema/schema.go

```diff
@@ -248,20 +245,22 @@ func (sh *Schema) isIDResolutionEnabled() bool {
 	return sh.enableIDResolution && slices.Contains(whutils.IdentityEnabledWarehouses, sh.warehouse.Type)
 }
 
-func (sh *Schema) UpdateLocalSchemaWithWarehouse(ctx context.Context) error {
-	sh.schemaInWarehouseMu.RLock()
-	defer sh.schemaInWarehouseMu.RUnlock()
-	return sh.updateLocalSchema(ctx, sh.schemaInWarehouse)
+func (sh *Schema) UpdateLocalSchemaWithWarehouse(ctx context.Context, warehouseSchema model.Schema) error {
+	return sh.updateSchema(ctx, warehouseSchema)
+}
+
+func (sh *Schema) UpdateLocalSchema(ctx context.Context) error {
+	return sh.updateSchema(ctx, sh.localSchema)
 }
 
-func (sh *Schema) UpdateLocalSchema(ctx context.Context, updatedSchema model.Schema) error {
-	return sh.updateLocalSchema(ctx, updatedSchema)
+func (sh *Schema) UpdateSchema(ctx context.Context, updatedSchema model.Schema) error {
```

Comment:
@cisse21 
Too many functions with same name 😅 
UpdateLocalSchemaWithWarehouse - Seems clear with func name.
UpdateLocalSchema - Seems to update in DB whatever is in memory?
UpdateSchema - ? Update in local and DB with whatever is passed?

Reply:
We already had many functions with similar names :/ didn't want to get into much more refactoring in this PR